mux/src/lib.rs

```diff
@@ -194,13 +195,21 @@ fn parse_buffered_data(pane: Weak<dyn Pane>, dead: &Arc<AtomicBool>, mut rx: Fil
                     // that we coalesce a full "frame" from an unoptimized
                     // TUI program
                     if action_size < buf.len() {
-                        if delay_ms > 0 {
+                        let next_update = match next_update {
+                            None => {
+                               next_update = Some(Instant::now() + delay);
+                               next_update.unwrap()
+                            },
+                            Some(next_update) => next_update,
+                        };
+                        let delay = next_update - Instant::now();
+                        if !delay.is_zero() {
```

Comment:
We need to avoid the `unwrap`. Also, Duration/Instant math can panic in some overflow situations, so I'd suggest using `checked_sub` to avoid that potential panic:

```suggestion
                        let poll_delay = match &deadline {
                            None => {
                               deadline.replace(Instant::now() + delay);
                               delay
                            },
                            Some(target) => target.checked_sub(Instant::now()).unwrap_or(Duration::ZERO)
                        };
                        if !poll_delay.is_zero() {                        
```

Reply:
Yes, that's what I was going for initially, but `checked_sub` takes a `Duration`, not an `Instant`. I looked again though and found the function `checked_duration_since` which does what we want.

It was a bit more concise to use `None` to represent an invalid `poll_delay`, rather than using zero, so that's what I went for in this change, but let me know if you prefer the other way.